### PR TITLE
feat: [PL-24394]: Help Panel

### PR DESCRIPTION
**Pull request by avinashmadhwani02**
You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- StyleLint: `retrigger stylelint`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- ImageSnapshot `retrigger ImageSnapshot`

### CODE REVIEW

File: packages/help-panel/src/ContentfulApi.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+class Contentful {
+  private client: ContentfulClientApi | undefined = undefined
+
+  getClient(): ContentfulClientApi {
+    if (this.client) {
+      return this.client
+    }
+
+    throw new Error('Please initialise Contentful before calling getClient')
```
[the remark this answers is not in the document]
Comment (avinashmadhwani02) [reply packages/help-panel/src/ContentfulApi.tsx]: done

File: packages/help-panel/src/HelpPanelContext.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  useEffect(() => {
+    if (Object.keys(referenceIdMap).length > 0) {
+      const contentId = referenceIdMap[referenceId]
+      if (contentId) {
+        setLoading(true)
+        Contentful.getClient()
+          .getEntries<T>({
+            'sys.id': contentId,
+            content_type: content_type,
+            include: 10
```
[the remark this answers is not in the document]
Comment (avinashmadhwani02) [reply packages/help-panel/src/HelpPanelContext.tsx:85]: it means fetching a maximum of 10 levels of nesting in response. Right now we have only 3 levels

For eg. Help panel -> Articles -> Image/Video

File: packages/help-panel/src/ContentfulApi.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (avinashmadhwani02) [reply packages/help-panel/src/ContentfulApi.tsx:19]: With your suggestion, we will have to pass `accessToken, space, environment` every time we call getClient, which should not be the case. 

We will be calling getClient multiple times through out the app, and we might not have access to all three parameters every time.

File: packages/help-panel/src/HelpPanelContext.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (mayankVermaHarness) [reply packages/help-panel/src/HelpPanelContext.tsx]: Thanks

**Review by mayankVermaHarness — APPROVED** [review]

**Pull request merged**